proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -277,13 +277,16 @@ private static String getFirstLine(String str)
         return pos == -1 ? str : str.substring( 0, pos );
     }
 
-    private ServerPing getPingInfo(String motd, int protocol)
-    {
-        return new ServerPing(
-                new ServerPing.Protocol( bungee.getName() + " " + bungee.getGameVersion(), protocol ),
-                new ServerPing.Players( listener.getMaxPlayers(), bungee.getOnlineCount(), null ),
-                motd, BungeeCord.getInstance().config.getFaviconObject()
-        );
+    private final ServerPing.Protocol cachedProtocol = new ServerPing.Protocol(
+    bungee.getName() + " " + bungee.getGameVersion(), bungee.getProtocolVersion()
+    );
+    private final ServerPing.Players cachedPlayers = new ServerPing.Players(
+    listener.getMaxPlayers(), bungee.getOnlineCount(), null
+    );
+    private final String cachedFavicon = BungeeCord.getInstance().config.getFaviconObject();
+
+    private ServerPing getPingInfo(String motd, int protocol) {
+    return new ServerPing(cachedProtocol, cachedPlayers, motd, cachedFavicon);
     }
```

Comment:
I believe some of these cached things an change with /greload
and indents missing here